src/components/graph/selectionToolbox/DeleteButton.vue

```diff
@@ -17,14 +19,14 @@ import Button from 'primevue/button'
 import { computed } from 'vue'
 import { useI18n } from 'vue-i18n'
 
+import { useSelectionState } from '@/composables/graph/useSelectionState'
 import { useCommandStore } from '@/stores/commandStore'
-import { useCanvasStore } from '@/stores/graphStore'
 
 const { t } = useI18n()
 const commandStore = useCommandStore()
-const canvasStore = useCanvasStore()
+const { selectedItems } = useSelectionState()
 
 const isDeletable = computed(() =>
-  canvasStore.selectedItems.some((x) => x.removable !== false)
+  selectedItems.value.some((x: any) => x.removable !== false)
```

Comment:
type this please.

Suggestion for fix:
```typescript
  interface RemovableItem {
    removable?: boolean
  }

  const isRemovableItem = (item: unknown): item is RemovableItem =>
    item != null && typeof item === 'object' &&
    ('removable' in item || true) // All objects are potentially removable

  const isDeletable = computed(() =>
    selectedItems.value
      .filter(isRemovableItem)
      .some((x) => x.removable !== false)
  )
```

@DrJKL can you think of a better way?

Reply:
for future PRs... if we're doing `isDeletable` a lot, we should consider putting it in the composable please.